pkg/appmanager/appManager.go

```diff
@@ -1309,6 +1314,18 @@ func (appMgr *Manager) syncRoutes(
 					rsCfg.deleteWhitelistCondition(dep.Name)
 				}
 			}
+			// Sort the rules and reset their ordinals
+			sortrules := func(rls *Rules, ordinal int) {
+				sort.Sort(sort.Reverse(*rls))
+				for _, v := range *rls {
```

Comment:
You can save this loop by updating the ordinal during the sort: https://play.golang.org/p/_DENN5qxR7e